### PR TITLE
types:add more unit tests in fsp

### PR DESCRIPTION
**Pull request by xiekeyi98**
### What problem does this PR solve? 
`fsp.go` has not unit tests, I added .

### What is changed and how it works?

Create new file `fsp_test.go` .

### Check List 

Tests 

 - Unit test

Side effects

 - Possible performance regression in tests(like CI)

### CODE REVIEW

File: types/fsp_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package types
```
Comment (zimulala) [inline types/fsp_test.go]: Please add the license at the beginning of a file.

Comment (xiekeyi98) [reply types/fsp_test.go]: done.

**Comment by xiekeyi98** [comment]
@zimulala  PTAL_(:з」∠)_

**Pull request merged**